### ISSUE TITLE
Keep expanded treeview first menu when click the 2nd menu

### ISSUE THREAD

**Issue by VJSash** [almasaeed2010/AdminLTE#1633]
Hi,
![acc](https://user-images.githubusercontent.com/31687058/30108617-934d33fe-9320-11e7-8fb3-16f8ceb2b43a.png)
Have to keep accordion menu open when selecting the another menu.. I have tried in app.js and css too. not much aware of this issue. I have to do like, once open the 1st menu, then choosing the another one, the first menu also keep open.

I had look #1476... but not working... thanks in advance...

**Comment by lenamtl** [comment]
Could you check if you have any error in the console right click inspect..
Do you have a live link so we can check

**Comment by VJSash** [comment]
lenamtl

there is no error in the console. dont have live link now..
I didnt change anything.. just change the labels in index.html and included the app.js in the bottom of index.html.
In the app.js 
changed the AdminLTE.tree function like below.. #1476

       $.AdminLTE.tree = function (menu) {
         var _this = this;
         var animationSpeed = $.AdminLTE.options.animationSpeed;
         $(document).off('click', menu + ' li a')
         .on('click', menu + ' li a', function (e) {
         //Get the clicked link and the next element
         var $this = $(this);
         var checkElement = $this.next();

        //If the menu is not visible open it
        if ((checkElement.is('.treeview-menu')) && (!checkElement.is(':visible'))) {
          //Get the parent menu
          var parent = $this.parents('ul').first();
          //Get the parent li
          var parent_li = $this.parent("li");

          //Open the target menu and add the menu-open class
          checkElement.slideDown(animationSpeed, function () {
            //Add the class active to the parent li
            checkElement.addClass('menu-open');
            parent_li.addClass('active');
            //Fix the layout in case the sidebar stretches over the height of the window
            _this.layout.fix();
          });
        }
        //if this isn't a link, prevent the page from being redirected
        if (checkElement.is('.treeview-menu')) {
          e.preventDefault();
        }
      });
  };

please give me the solution...

**Comment by lenamtl** [comment]
Ok I will test it.
Could you check the link of your file to make sure you are linking to app.js not app.min.js
to make sure you are using the edited file...

**Comment by VJSash** [comment]
Yes..  linking app.js in the bottom of file.

added like this...


<script src="dist/js/demo.js"></script>
<script src="dist/js/app.js"></script>

using AdminLTE v2.4.0-rc version.
please do the needful.. thank you...

**Comment by lenamtl** [comment]
You have 3 choice 

Using v 2.3 and the code provided https://github.com/almasaeed2010/AdminLTE/issues/1476

If you are using v2.4 this seems to be supported by default so you don't need to change anything.
(by the way 2.4 have different js files no more apps.js)

If you have upgrade from 2.3 to 2.4
see the upgrade guide https://adminlte.io/docs/2.4/upgrade-guide 
and you will need to add to` data-api="tree" data-accordion=0 `to your sidebar menu.

I'm still using the v2.3 because I have heavily modified it...

**Comment by VJSash** [comment]
using AdminLTE v2.4.0-rc version.
not working... please check this file.

[index.txt](https://github.com/almasaeed2010/AdminLTE/files/1286782/index.txt)

**Comment by REJack** [comment]
You need to add `data-api="tree" data-accordion=0` to `<ul class="sidebar-menu" data-widget="tree">` like @lenamtl wrote.

It need to look like this:
```html
<ul class="sidebar-menu" data-widget="tree" data-api="tree" data-accordion=0>
```

You can look at this modded example [index.txt](https://github.com/almasaeed2010/AdminLTE/files/1286981/index.txt) too.

**Comment by VJSash** [comment]
fantastic!!!. its working.. thank you so much. @lenamtl  and @REJack

**Comment by usmanbasharmal123** [comment]
I face with same problem but I solved by the below code  (AdminLTE-3.0.0-beta.2)
<ul class="nav nav-pills nav-sidebar flex-column" data-widget="treeview" role="menu" data-accordion="true">
but when you click on dashboard if it is not in treeview then it not going to close or collapse the menu so you have to put it in the li->ul->li gag like treeview to wok as will.
